### ISSUE TITLE
Consider force restarting the server for very long running VS Code instances

### ISSUE THREAD

**Issue by vinistock** [Shopify/vscode-ruby-lsp#856]
Some users never close or restart VS Code unless they absolutely have to. This means that - unless they modify a watched file like Gemfile.lock - they will never trigger the custom bundle logic and never try to update the server.

It would be great if we could detect that the VS Code window is idle, silently shutdown the server and then re-initialize it once there's activity again. Not sure if VS Code provides any APIs for this.

**Comment by st0012** [comment]
Do we have data showing how many users this could be benefiting, and how much degraded experience the problem is causing?

While I understand it's to help users get a better experience, the extra infra we need to add and maintain is concerning. We'll need to add this on both extension and the server, find ways to test them, and risking further-complicating the activation logic. So I'd avoid solving this unless we can easily justify it.

**Comment by vinistock** [comment]
I agree that we need to understand how big of an issue this is - especially because the idle detection could lead to weird scenarios.

But I don't understand what you mean by extra infra and why the server would be involved. We could just listen for changes and consider that to be an idle indicator, which would clear a timer set to stop the server. 

Something like this

```typescript
// Automatically start the server when it became idle 
vscode.workspace.onDidChangeTextDocument((_event) => {
  if (this.client.state === State.Idle) {
    this.client?.start();
  }

  clearTimeout(this.idleTimer);
});

async start() {
  this.idleTimer = setTimeout(() => {
    await this.client.stop();
    this.client.state = State.Idle;
  }, ONE_HOUR);
}
```
